test/e2e/basic_ops_test.go

```diff
@@ -0,0 +1,123 @@
+)
+
+var _ = Describe("basic", func() {
+	var (
+		specName       = "basic-ops"
+		namespace      *corev1.Namespace
+		cancelWatches  context.CancelFunc
+		bmcUser        string
+		bmcPassword    string
+		bmcAddress     string
+		bootMacAddress string
+		vmName         string
+	)
+	const (
+		rebootAnnotation   = "reboot.metal3.io"
+		poweroffAnnotation = "reboot.metal3.io/poweroff"
+	)
+	BeforeEach(func() {
+		bmcUser = e2eConfig.GetVariable("BMC_USER")
+		bmcPassword = e2eConfig.GetVariable("BMC_PASSWORD")
+		bmcAddress = e2eConfig.GetVariable("BMC_ADDRESS")
+		bootMacAddress = e2eConfig.GetVariable("BOOT_MAC_ADDRESS")
+		vmName = e2eConfig.GetVariable("VM_NAME")
+
+		namespace, cancelWatches = framework.CreateNamespaceAndWatchEvents(ctx, framework.CreateNamespaceAndWatchEventsInput{
+			Creator:   clusterProxy.GetClient(),
+			ClientSet: clusterProxy.GetClientSet(),
+			Name:      fmt.Sprintf("%s-%s", specName, util.RandomString(6)),
+			LogFolder: artifactFolder,
+		})
+	})
+
+	It("should power cycle a newly created BMH", func() {
+		By("creating a secret with BMH credentials")
+		bmcCredentials := corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "bmc-credentials",
+				Namespace: namespace.Name,
+			},
+			StringData: map[string]string{
+				"username": bmcUser,
+				"password": bmcPassword,
+			},
+		}
+		err := clusterProxy.GetClient().Create(ctx, &bmcCredentials)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a BMH")
+		bmh := metal3api.BareMetalHost{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      specName + "-powercycle",
+				Namespace: namespace.Name,
+				Annotations: map[string]string{
+					"inspect.metal3.io": "disabled",
+				},
+			},
+			Spec: metal3api.BareMetalHostSpec{
+				Online: true,
+				BMC: metal3api.BMCDetails{
+					Address:         bmcAddress,
+					CredentialsName: "bmc-credentials",
+				},
+				BootMode:       metal3api.Legacy,
+				BootMACAddress: bootMacAddress,
+			},
+		}
+		err = clusterProxy.GetClient().Create(ctx, &bmh)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("waiting for the BMH to be in registering state")
+		WaitForBmhInState(ctx, WaitForBmhInStateInput{
+			Client: clusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  metal3api.StateRegistering,
+		}, e2eConfig.GetIntervals("inspection", "wait-registering")...)
+
+		By("waiting for the BMH to become available")
+		WaitForBmhInState(ctx, WaitForBmhInStateInput{
+			Client: clusterProxy.GetClient(),
+			Bmh:    bmh,
+			State:  metal3api.StateAvailable,
+		}, e2eConfig.GetIntervals("inspection", "wait-available")...)
+
+		By("testing the reboot BMH")
+		capm3_e2e.AnnotateBmh(ctx, clusterProxy.GetClient(), bmh, rebootAnnotation, pointer.String("{\"force\": true}"))
+		waitForVmsState([]string{vmName}, shutoff, e2eConfig.GetIntervals(specName, "wait-vm-state")...)
+		waitForVmsState([]string{vmName}, running, e2eConfig.GetIntervals(specName, "wait-vm-state")...)
```

Comment:
This is a good way to ensure that it is really doing what we think. However, it is also impossible to do with real hardware. I was hoping we could do these e2e tests so that they can run also against real hardware in various configurations.
Do you think there is any chance that we could check these things in a different way? Maybe [PoweredOn](https://github.com/metal3-io/baremetal-operator/blob/7c007b6ae4efe36d2f313a0573d45804b03954b0/apis/metal3.io/v1alpha1/baremetalhost_types.go#L790) could work if the reboot is not too fast?